pkg/koordlet/resourceexecutor/executor.go

```diff
@@ -47,16 +53,31 @@ type ResourceUpdateExecutorImpl struct {
 	ResourceCache     *cache.Cache
 	Config            *Config
 
-	onceRun   sync.Once
-	gcStarted bool
+	onceRun       sync.Once
+	gcStarted     bool
+	eventRecorder record.EventRecorder
+	node          *corev1.Node
 }
 
 var singleton = &ResourceUpdateExecutorImpl{
 	ResourceCache: cache.NewCacheDefault(),
 	Config:        Conf,
 }
 
+var recorderSingleton record.EventRecorder
+var nodeSingleton *corev1.Node
+
+func NewResourceExecutorEventRecorder(schema *apiruntime.Scheme, kubeClient kubernetes.Interface, nodeName string, node *corev1.Node) {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(&clientcorev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(schema, corev1.EventSource{Component: "koordlet-resource-executor", Host: nodeName})
+	recorderSingleton = recorder
+	nodeSingleton = node
+}
+
 func NewResourceUpdateExecutor() ResourceUpdateExecutor {
+	singleton.eventRecorder = recorderSingleton
```

Comment:
Why separate the initialization of the executor and the recorder but not new the event recorder in the singleton's definition?

Reply:
OK , fixed it.

Reply:
OK, fixed it, PTAL